subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/TestFile.java

```diff
@@ -709,6 +709,12 @@ public TestFile createFile() {
         return this;
     }
 
+    public TestFile makeUnreadable() {
+        setReadable(false, false);
```

Comment:
Could probably do `Files.setPosixFilePermissions(toPath(), Collections.emptySet())` as well, if we want to be really hard-core